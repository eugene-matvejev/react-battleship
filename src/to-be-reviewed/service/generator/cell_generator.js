import CellModel from '../../model/cell_model';

/**
 * @param {Number} x
 * @param {Number} y
 *
 * @returns {CellModel}
 */
export const generateCell = (x, y) => new CellModel(`${String.fromCharCode(96 + x).toUpperCase()}${y}`);
