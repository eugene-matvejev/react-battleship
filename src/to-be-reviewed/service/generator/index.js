export { generateBattlefield } from './battlefield_generator';
export { generateCell } from './cell_generator';
export { generateGame } from './game_generator';
