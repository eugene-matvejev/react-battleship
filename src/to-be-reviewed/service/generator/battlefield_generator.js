import BattlefieldModel from '../../model/battlefield_model';
import { generateCell } from './cell_generator';

/**
 * @param {Number} size
 *
 * @returns {BattlefieldModel}
 */
export const generateBattlefield = (size) => {
    const cells = [];
    for (let x = 1; x <= size; x++) {
        for (let y = 1; y <= size; y++) {
            const cell = generateCell(x, y);

            cells.push(cell);
        }
    }

    const model = new BattlefieldModel();
    model.size = size;
    model.setCells(cells);

    return model;
};
