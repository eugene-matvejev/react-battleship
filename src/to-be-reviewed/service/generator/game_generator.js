import CellModel from '../../model/cell_model';
import GameModel from '../../model/game_model';
import PlayerModel from '../../model/player_model';
import { generateBattlefield } from './battlefield_generator';

/**
 * @param {Number} players
 * @param {Number} size
 *
 * @returns {GameModel}
 */
export const generateGame = (players, size) => {
    const model = new GameModel();

    for (let i = 0; i < players; i++) {
        const battlefield = generateBattlefield(size);
        const player = new PlayerModel();
        if ((i + 1) === players) {
            player.addSequence(PlayerModel.getHumanFlag())
        }

        battlefield.setPlayer(player);
        model.addBattlefield(battlefield)
    }

    injectPlayerShips(model);

    return model;
}

/**
 * @param {GameModel} model
 */
const injectPlayerShips = (model) => {
    const shipCoordinates = ['A1', 'A2', 'A3', 'A5', 'C1', 'C2', 'C5'];
    const deadCoordinates = ['A3', 'A4', 'B1', 'B2', 'B3', 'B4'];
    const { ship, dead } = CellModel.flags;

    for (const battlefield of model.getBattlefields()) {
        if (!battlefield.getPlayer().isHumanControlled()) {
            continue;
        }

        shipCoordinates.forEach((v) => callback(v, ship, battlefield));
        deadCoordinates.forEach((v) => callback(v, dead, battlefield));
    }
}

const callback = (c, seq, battlefield) => {
    const cell = battlefield.getCellByCoordinate(c);

    if (undefined !== cell) {
        cell.addSequence(seq);
    }
};
