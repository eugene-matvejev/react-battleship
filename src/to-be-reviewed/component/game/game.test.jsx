import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Game from './game';
import { generateGame } from '../../service/generator';

configure({ adapter: new Adapter() });

describe(`<Game/>`, () => {
    const model = generateGame(2, 1);

    describe(`rendering`, () => {
        it('with default/required props', () => {
            shallow(<Game model={model} />);
        });

        it('match snapshot', () => {
            const c = shallow(<Game model={model} />);

            expect(c).toMatchSnapshot();
        });
    });
});
