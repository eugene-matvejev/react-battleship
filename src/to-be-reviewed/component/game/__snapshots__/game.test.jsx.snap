// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game/> rendering match snapshot 1`] = `
<div
  className="component game "
>
  <div
    className="col-md-6"
    key="0"
  >
    <Player
      avatarSrc="/assets/img/avatar-placeholder.png"
      byteSeq={0}
      className=""
      id="undefined"
      name="undefined"
      score={0}
    />
    <Battlefield
      className=""
      model={
        BattlefieldModel {
          "byteSeq": 0,
          "cellsIndexedByCoordinate": Object {
            "A1": CellModel {
              "byteSeq": 0,
              "coordinate": "A1",
            },
          },
          "cellsIndexedById": Object {},
          "id": undefined,
          "player": PlayerModel {
            "avatarSrc": "/assets/img/avatar-placeholder.png",
            "byteSeq": 0,
            "id": "undefined",
            "name": "undefined",
            "score": 0,
          },
          "size": 1,
        }
      }
    />
  </div>
  <div
    className="col-md-6"
    key="1"
  >
    <Player
      avatarSrc="/assets/img/avatar-placeholder.png"
      byteSeq={1}
      className=""
      id="undefined"
      name="undefined"
      score={0}
    />
    <Battlefield
      className=""
      model={
        BattlefieldModel {
          "byteSeq": 0,
          "cellsIndexedByCoordinate": Object {
            "A1": CellModel {
              "byteSeq": 2,
              "coordinate": "A1",
            },
          },
          "cellsIndexedById": Object {},
          "id": undefined,
          "player": PlayerModel {
            "avatarSrc": "/assets/img/avatar-placeholder.png",
            "byteSeq": 1,
            "id": "undefined",
            "name": "undefined",
            "score": 0,
          },
          "size": 1,
        }
      }
    />
  </div>
</div>
`;
